advanced-creation.md

```diff
@@ -111,3 +112,119 @@ const unicorn = got.create(settings);
 // Same as:
 const unicorn = got.extend({headers: {unicorn: 'rainbow'}});
 ```
+
+### Merging instances
+
+#### got.mergeInstances(instanceA, instanceB, ...)
+
+Merges many instances into a single one:
+- options are merged using [`got.mergeOptions()`](readme.md#gotmergeoptionsparentoptions-newoptions) (+ hooks are merged too),
+- handlers are stored in an array.
+
+## Usage
+
+#### Denying redirects that lead to other sites than specified
+
+```js
+const controlRedirects = got.create({
+	options: got.defaults.options,
+	handler: (options, next) => {
+		const promiseOrStream = next(options);
+		return promiseOrStream.on('redirect', resp => {
+			const host = new URL(resp.url).host;
+			if (options.allowedHosts && !options.allowedHosts.includes(host)) {
+				promiseOrStream.cancel(`Redirection to ${host} is not allowed`);
+			}
+		});
+	}
+});
+```
+
+#### Limiting download & upload (in case your machine's got a little amount of RAM)
+
+```js
+const limitDownloadUpload = got.create({
+    options: got.defaults.options,
+    handler: (options, next) => {
+        let promiseOrStream = next(options);
+        if (typeof options.downloadLimit === 'number') {
+            promiseOrStream.on('downloadProgress', progress => {
+        		if (progress.transferred > options.downloadLimit && progress.percent !== 1) {
+        			promiseOrStream.cancel(`Exceeded the download limit of ${options.downloadLimit} bytes`);
+        		}
+        	});
+        }
+
+        if (typeof options.uploadLimit === 'number') {
+            promiseOrStream.on('uploadProgress', progress => {
+        		if (progress.transferred > options.uploadLimit && progress.percent !== 1) {
+        			promiseOrStream.cancel(`Exceeded the upload limit of ${options.uploadLimit} bytes`);
+        		}
+        	});
+        }
+        
+        return promiseOrStream;
+    }
+});
+```
+
+#### No user agent
+
+```js
+const noUserAgent = got.extend({
+	headers: {
+		'user-agent': null
+	}
+});
+```
+
+#### Custom endpoint
+
+```js
+const httpbin = got.extend({
+	baseUrl: 'https://httpbin.org/'
+});
+```
+
+#### Signing requests
+
+```js
+const crypto = require('crypto');
+const getMessageSignature = (data, secret) => crypto.createHmac('sha256', secret).update(data).digest('hex').toUpperCase();
+const signRequest = got.extend({
+	hooks: {
+		beforeRequest: [
+			options => {
+				options.headers['sign'] = getMessageSignature(options.body || '', process.env.SECRET);
+			}
+		]
+	}
+});
+```
+
+If these ^^^ are different modules and you don't want to rewrite them, use `got.mergeInstances()`.
+
```

Comment:
There should be a header here called `#### Putting it all together` or something.